### ISSUE TITLE
Core JS: handleError interferes with periodic Poller

### ISSUE THREAD

**Issue by frueh190** [primefaces/primefaces#14890]
### Describe the bug

PrimeFaces 14.0.14 introduced a new call to PrimeFaces.ajax.Utils.handleError for AJAX errors:

```
core.ajax.js:
    $(document).on('pfAjaxError', function(e, xhr, settings, error){
        // this is very likely a connection error
        PrimeFaces.ajax.Utils.handleError("", "AJAX failure");
    });
```

It gets also called when a periodic poller is unable to complete its last cycle, because the user leaves the page.

```
template.xhtml:
<p:poll interval="3" timeout="60000" widgetVar="poll" global="false"
    partialSubmit="true" listener="#{viewBean.checkForUpdates}" onerror="window.location.reload()"/>
```

To reproduce the AJAX error it helps to delay the response from the view bean:

```
ViewBean.java
public void checkForUpdates() throws Exception {
    Thread.sleep(100000);
}
```

JavaScript output (without error-page in web.xml):
```
No ajaxExceptionHandler or error page found for '' defined! [core.js.xhtml:25:9]
AJAX failure [core.js.xhtml:25:9]
Request return with error:error. [core.js.xhtml:25:9]
```

First question: Is this an intended behaviour? It is obviously NOT a connection error, when the last poller cycle cannot be completed because the user simply changes the page / clicks a button.

Second question: How can I define an appropriate error handling? I have a catch-all error-page in web.xml. And I don't hava an AjaxExceptionHandler. So the user gets redirected to the catch-all error.xhtml and is confused, because in his opinion nothing bad has happened (which is true).

Third question: How do I define an AjaxExceptionHandler? The function handleError("", "AJAX failure") is called with an empty string as first argument. With an empty parameter errorName I cannot define a specific AjaxExceptionHandler. How do I define a global AjaxExceptionHandler instead?

### Expected behavior

When a poller cycle cannont be completed, the PrimeFaces.ajax.Utils.handleError should not be called. Because it didn't get called before PrimeFaces version 15.0.14 an such a design change is not suitable to be introduced in a minor version update.

### PrimeFaces edition

None

### PrimeFaces version

15.0.14

### JSF implementation

MyFaces

### JSF version

4.1.3

### Java version

21

**Comment by tandraschko** [comment]
When the poller hits a error, i think error handling should be called.

If you have a global error page, a redirect to it must happen via JS.
Please give 15.x or 16-SNAPSHOT a try or provide a reproducer.

**Comment by melloware** [comment]
Yes try 15.0.15

**Comment by wlhUser** [comment]
@frueh190 you saved my weekend.
I was debugging for hours on a problem caused by this very issue.

In my application users can trigger an Ajax-request (which might take a few seconds), but if they do not want to wait until it is finished, they can click somewhere else and leave the current page. However, this is now considered as an error, and the application navigates to the error page without showing any error or exception.
In my case this happens with PF-15.0.15

**Comment by tandraschko** [comment]
on page unload, we actually abort all AJAX requests:

```
const unloadEvent = ("onpagehide" in window) ? "pagehide" : "unload";
 $(window).on(unloadEvent, ()  => PrimeFaces.ajax.Queue.abortAll());
```

maybe, just maybe, our AJAX core needs:
if the AJAX error caused by abort -> skip all error handling -> also dont call onerror attr

**Comment by tandraschko** [comment]
something like:

https://github.com/primefaces/primefaces/blob/9b04eaf0fc5458a2af7dd375c01252841dce80c1/primefaces/src/main/frontend/packages/core/src/core/core.ajax.ts#L930

```
    if (status === 'abort') {
        return; // intentionally aborted, not a real error
    }
```

we would need some testing

**Comment by frueh190** [comment]
I'm struggling to create a reproducer. Because the root cause of the problem lies somewhere deeper in my customer's application. The JavaScript error 

`Request return with error:error.`

already shows up in the Browser Console with PrimeFaces 15.0.13. But it was swallowed and no one took notice. With PrimeFaces 15.0.14 and 15.0.15 it is no longer swallowed and the user is redirected to the error.xhtml.

When I create a primefaces-test-App with just a Poller and a Link which interrupts the poller midways, everythin is fine.

**Comment by tandraschko** [comment]
maybe you can debug the JS on your real app and see if status == abort in our ajax core?

**Comment by frueh190** [comment]
[primefaces-test-14890.zip](https://github.com/user-attachments/files/27530528/primefaces-test-14890.zip)

**Comment by frueh190** [comment]
Ooookay. Maybe we're having a browser feature here? With Firefox, on the "old" page the JavaScript error reads **error** (you need to hit the check box "Preserve log" to see the message):

`Request return with error:error.`

This leads to a redirect to error.xhtml -> not desired

With Microsoft Edge the JavaScript error reads **abort** (not always, but most of the time):

`Request return with error:abort.`

Apart from the message in the console nothing else happens -> fine

I will attach my primefaces-test-App. To reproduce the error simple hit the Link "Change page now!" as soon as the poller logs the message "Change page now!". Test with Firefox. Don't open two different browsers at the same time...

**Comment by melloware** [comment]
@frueh190 I suspect Chrome and Edge are doing the right thing so its Firefox only having the issue?

**Comment by frueh190** [comment]
With Firefox I can reproduce the error every single time.

With Chrome and Edge the error occurs consistently on the **very first** request after a browser restart. When I go "back" in the browser and try again, everything is working perfectly. When I hit the Link in my Test App **before** the poller starts for the first time, the error will **never** occur.

**Comment by melloware** [comment]
cc @christophs78 this might be related to the nightly build failures?

**Comment by melloware** [comment]
adding @lprimak as i think this was also due to recent changes

**Comment by lprimak** [comment]
It's probably Firefox that's going the right thing :). 
Anyway you can try a workaround https://github.com/flowlogix/website/blob/main/content/js/apps/ajax-error-handler.js

See if that helps

**Comment by melloware** [comment]
in Firefox this is the XHR:

```js
XHR {
  "readyState": 0,
  "pfSettings": {
    "url": "/test.xhtml",
    "type": "POST",
    "isLocal": false,
    "global": false,
    "processData": true,
    "async": true,
    "contentType": "application/x-www-form-urlencoded; charset=UTF-8",
    "accepts": {
      "*": "*/*",
      "text": "text/plain",
      "html": "text/html",
      "xml": "application/xml, text/xml",
      "json": "application/json, text/javascript",
      "script": "text/javascript, application/javascript, application/ecmascript, application/x-ecmascript"
    },
    "contents": {
      "xml": {},
      "html": {},
      "json": {}
    },
    "responseFields": {
      "xml": "responseXML",
      "text": "responseText",
      "json": "responseJSON"
    },
    "converters": {
      "text html": true
    },
    "flatOptions": {
      "url": true,
      "context": true
    },
    "jsonp": "callback",
    "cache": false,
    "dataType": "xml",
    "portletForms": null,
    "source": "frmTest:j_idt9",
    "data": "jakarta.faces.partial.ajax=true&jakarta.faces.source=frmTest%3Aj_idt9&jakarta.faces.partial.execute=%40all&frmTest%3Aj_idt9=frmTest%3Aj_idt9&frmTest=frmTest&jakarta.faces.ViewState=4064983870010870080%3A180276620620640061",
    "dataTypes": [
      "xml"
    ],
    "crossDomain": false,
    "hasContent": true
  },
  "pfArgs": {},
  "status": 0,
  "statusText": "error"
}
```

**Comment by melloware** [comment]
ok this is helpful...

This is a long-standing browser difference in how aborted `XMLHttpRequest` objects surface events, and jQuery inherits that behavior.

In Firefox:

* `xhr.abort()` typically results in:

  * `status = 0`
  * `statusText = "error"` or `"abort"`
  * jQuery triggers `.fail()`

In Chrome:

* `xhr.abort()` often results in:

  * request cancellation without triggering `.fail()`
  * or only triggering `.always()`
  * depending on timing and jQuery version

So this is not really a PrimeFaces issue — it is browser/XHR event behavior divergence.

The safest cross-browser approach is:

1. Handle `abort` explicitly
2. Use `.always()` for cleanup
3. Track manual aborts yourself

Example:

```javascript
let aborted = false;

const xhr = $.ajax({
    url: '/test.xhtml',
    method: 'POST'
})
.fail((jqXHR, textStatus, errorThrown) => {
    console.log('FAIL', textStatus, errorThrown);

    if (aborted || textStatus === 'abort') {
        console.log('Request was aborted');
        return;
    }

    console.log('Actual AJAX failure');
})
.always((jqXHR, textStatus) => {
    console.log('ALWAYS', textStatus);
});

function cancelRequest() {
    aborted = true;
    xhr.abort();
}
```

You can also normalize behavior globally:

```javascript
$(document).ajaxError((event, jqXHR, settings, thrownError) => {
    if (jqXHR.status === 0) {
        console.log('Request aborted or network interruption');
        return;
    }

    console.log('Real AJAX error');
});
```

But note:

* `status === 0` can also mean:

  * CORS failure
  * network disconnect
  * navigation interruption
  * blocked request

So if you specifically care about *intentional* aborts, maintaining your own flag is the most reliable solution.

Also, newer browser APIs (`fetch` + `AbortController`) behave much more consistently across browsers than legacy XHR/jQuery abort handling.

**Comment by melloware** [comment]
OK the real issue is not `abort` but on page navigation the browser tears down the current page and stops all Javascript.  Chrome handles this gracefully and does not send a `jquery.fail` however Firefox does and reports it as `status===0` which can't be discerned from other types of `0` like CORS/Network Failure.

The solution in my PR is working in Firefox, Chrome and Edge.

**Comment by lprimak** [comment]
Awesome @melloware you are the best

**Comment by frueh190** [comment]
Thanks for digging deep into the JavaScript / Browser jungle!

**Comment by wlhUser** [comment]
@melloware I'm glad you fixed this. Thank you very much for your effort

**Comment by christophs78** [comment]
@melloware - great work, nightly build is green again

**Comment by melloware** [comment]
Awesome!!
